chain/network/src/network_protocol/mod.rs

```diff
@@ -284,23 +289,39 @@ impl PeerMessage {
     pub(crate) fn serialize(&self, enc: Encoding) -> Vec<u8> {
         match enc {
             Encoding::Borsh => borsh_::PeerMessage::from(self).try_to_vec().unwrap(),
-            Encoding::Proto => proto::PeerMessage::from(self).write_to_bytes().unwrap(),
+            Encoding::Proto => {
+                let mut msg = proto::PeerMessage::from(self);
+                let propagator = NodePropagator::new();
+                let cx = Span::current().context();
+                msg.trace_context = propagator.inject_trace_context(&cx);
+                msg.write_to_bytes().unwrap()
+            }
         }
     }
 
     pub(crate) fn deserialize(
         enc: Encoding,
         data: &[u8],
     ) -> Result<PeerMessage, ParsePeerMessageError> {
+        let span = tracing::trace_span!(target: "network", "deserialize").entered();
         Ok(match enc {
             Encoding::Borsh => (&borsh_::PeerMessage::try_from_slice(data)
                 .map_err(ParsePeerMessageError::BorshDecode)?)
                 .try_into()
                 .map_err(ParsePeerMessageError::BorshConv)?,
-            Encoding::Proto => (&proto::PeerMessage::parse_from_bytes(data)
-                .map_err(ParsePeerMessageError::ProtoDecode)?)
-                .try_into()
-                .map_err(ParsePeerMessageError::ProtoConv)?,
+            Encoding::Proto => {
+                let proto_msg: proto::PeerMessage = proto::PeerMessage::parse_from_bytes(data)
+                    .map_err(ParsePeerMessageError::ProtoDecode)?;
+                if proto_msg.trace_context.is_some() {
+                    let propagator = NodePropagator::new();
+                    if let Ok(extracted_span_context) =
+                        propagator.extract_span_context(&proto_msg.trace_context)
```

Comment:
shouldn't the parsing error be logged here?

Reply:
I'm worried about spamming the logs.
I wish I had `LOG_EVERY_N_SECONDS()`.